applications/views/applications.py

```
@@ -492,6 +494,23 @@ def put(self, request, pk):
                     status=status.HTTP_403_FORBIDDEN,
                 )
 
+            # Check if any blocking flags are on the case
+            blocking_flags = (
+                get_flags(application.get_case())
+                .filter(status=FlagStatuses.ACTIVE, blocks_approval=True)
+                .order_by("name")
+                .values_list("name", flat=True)
+            )
+            if blocking_flags:
+                return JsonResponse(
```

I think this should raise a forbidden exception instead, like so `raise exceptions.PermissionDenied([f"{strings.Applications.Finalise.Error.BLOCKING_FLAGS}{','.join(list(blocking_flags))}"])`